tests/port_dpb.py

```diff
@@ -0,0 +1,202 @@
+        self.set_alias(child_ports[0].get_alias().rsplit(',',1)[0])
+        self.set_index(child_ports[0].get_index())
+        lanes =[]
+        for cp in child_ports:
+            for l in cp.get_lanes():
+                lanes.append(l)
+        self.set_lanes(lanes)
+
+
+    def port_split(self, child_ports):
+        if child_ports == 1:
+            return self
+        child_port_list = []
+        port_num = self.get_port_num()
+        num_lanes = len(self._lanes)
+        offset = num_lanes/child_ports;
+        lanes_per_child = offset
+        for i in range(child_ports):
+            child_port_num = port_num + (i * offset)
+            child_port_name = "Ethernet%d"%(child_port_num) 
+            child_port_alias = "Eth%d/%d"%(port_num, child_port_num)
+            child_port_lanes = []
+            for j in range(lanes_per_child):
+                child_port_lanes.append(self._lanes[(i*offset)+j])
+            child_port_speed = self._speed/child_ports
+            child_port_index = self._index
+
+            child_port = Port(self._dvs, child_port_name)
+            child_port.set_alias(child_port_alias)
+            child_port.set_speed(child_port_speed)
+            child_port.set_lanes(child_port_lanes)
+            child_port.set_index(child_port_index)
+            child_port_list.append(child_port)
+        return child_port_list
+
+    def delete_from_config_db(self):
+        self._cfg_db_ptbl._del(self.get_name())
+        self._oid = None
+
+    def sync_from_config_db(self):
+        (status, fvs) = self._cfg_db_ptbl.get(self.get_name())
+        assert status == True
+        fvs_dict = self.get_fvs_dict(fvs)
+        self.set_alias(fvs_dict['alias'])
+        self.set_speed(int(fvs_dict['speed']))
+        self.set_lanes(list(fvs_dict['lanes'].split(",")))
+        self.set_index(int(fvs_dict['index']))
+
+    def write_to_config_db(self):
+        lanes_str = self.get_lanes_db_str() 
+        index_str = str(self.get_index())
+        alias_str = self.get_alias() 
+        speed_str = str(self.get_speed())
+        fvs = swsscommon.FieldValuePairs([("alias", alias_str),
+                                          ("lanes", lanes_str),
+                                          ("speed", speed_str),
+                                          ("index", index_str)])
+        self._cfg_db_ptbl.set(self.get_name(), fvs)
+
+    def get_fvs_dict(self, fvs):
+        fvs_dict = {}
+        for fv in fvs:
+            fvs_dict.update({fv[0]:fv[1]})
+        return fvs_dict
+
+    def exists_in_config_db(self):
+        (status, fvs) = self._cfg_db_ptbl.get(self.get_name())
+        return status
+
+    def exists_in_app_db(self):
+        (status, fvs) = self._app_db_ptbl.get(self.get_name())
+        return status
+
+    def exists_in_asic_db(self):
+        if self._oid is None:
+            counter_redis_conn = redis.Redis(unix_socket_path=self._dvs.redis_sock, db=swsscommon.COUNTERS_DB)
```

Comment:
change this part to be in common class member initialization?

Reply:
Yes, thats how I started, but I faced some issue on keeping the redis connection intact, hence moved it here.